ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ip-igmp-snooping.j2

```diff
@@ -1,26 +1,61 @@
-{% if ip_igmp_snooping is defined and ip_igmp_snooping is not none %}
+{% if ip_igmp_snooping is arista.avd.defined %}
 
 ## IP IGMP Snooping
 
 ### IP IGMP Snooping Summary
 
-{%     if ip_igmp_snooping.globally_enabled is defined and ip_igmp_snooping.globally_enabled == false %}
-IGMP snooping is globally disabled
+| Global Settings | Values |
+| --------------- | ------ |
+{%     if ip_igmp_snooping.globally_enabled is arista.avd.defined(false) %}
+| IGMP snooping | Disabled |
 {%     else %}
-IGMP snooping is globally enabled.
-
-{%         if ip_igmp_snooping.vlans is arista.avd.defined %}
-
-| VLAN | IGMP Snooping |
-| --- | --------------- |
-{%             for vlan in ip_igmp_snooping.vlans | arista.avd.natural_sort %}
-{%                 if ip_igmp_snooping.vlans[vlan].enabled is defined and ip_igmp_snooping.vlans[vlan].enabled == false %}
-| {{ vlan }} | disabled |
-{%                 elif ip_igmp_snooping.vlans[vlan].enabled is defined and ip_igmp_snooping.vlans[vlan].enabled == true %}
-| {{ vlan }} | enabled |
-{%                 endif %}
+| IGMP snooping | Enabled |
+| Fast-leave | {{ ip_igmp_snooping.fast_leave | arista.avd.default(true) }} |
+{%         if ip_igmp_snooping.interface_restart_query is arista.avd.defined %}
+| Interface restart query | {{ ip_igmp_snooping.interface_restart_query }} |
+{%         endif %}
+| Proxy | {{ ip_igmp_snooping.proxy | arista.avd.default(false) }} |
+{%         if ip_igmp_snooping.restart_query_interval is arista.avd.defined %}
+| Restart query interval | {{ ip_igmp_snooping.restart_query_interval }} |
+{%         endif %}
+{%         if ip_igmp_snooping.robustness_variable is arista.avd.defined %}
+| Robustness variable | {{ ip_igmp_snooping.robustness_variable }} |
+{%         endif %}
+{%         if ip_igmp_snooping.querier is arista.avd.defined %}
+
+| Global Querier Settings | Values |
+| ----------------------- | ------ |
+{%             for querier_setting in ip_igmp_snooping.querier | arista.avd.natural_sort if ip_igmp_snooping.querier[querier_setting] is arista.avd.defined %}
```

Comment:
This method is a bit lazy :) We need to ensure proper wording of settings and put abreviations in upper case.

Reply:
I am going with Querier Setting because all the knobs are single valued.